querybook/server/lib/query_executor/clients/hive.py

```diff
@@ -116,27 +116,31 @@ def tracking_url(self):
         return self._tracking_url
 
     def _update_percent_complete(self, poll_result):
-        task_status = poll_result.taskStatus
-        if task_status:
-            try:
-                parsed_task_status = json.loads(task_status)
-                map_reduce_stages = [
-                    stage
-                    for stage in parsed_task_status
-                    if stage.get("taskType", "") == "MAPRED"
-                ]
-                if len(map_reduce_stages) > 0:
-                    stage_sum = sum(
-                        map(
-                            lambda stage: stage.get("mapProgress", 0)
-                            + stage.get("reduceProgress", 0),
-                            map_reduce_stages,
+        if poll_result.progressUpdateResponse and poll_result.progressUpdateResponse.progressedPercentage:
```

Comment:
This if check might error out for hive 1.2. Can you use `getattr` to be on the safer side?
```
if getattr(poll_result, 'progressUpdateResponse', None):
    update_resp = poll_result.progressUpdateResponse
    percent_complete = update_resp.progressedPercentage if getattr(update_resp.progressedPercentage, None) else 0
    ...
```
also removed the percentage check in the if to prevent going to the else part if progressPercentage is 0

Reply:
changed